gmso/external/convert_hoomd.py

```diff
@@ -0,0 +1,881 @@
+            "container": hoomd.md.pair.LJ,
+            "parser": _parse_lj,
+        },
+        "BuckinghamPotential": {
+            "container": hoomd.md.pair.Buckingham,
+            "parser": _parse_buckingham,
+        },
+        "MiePotential": {"container": hoomd.md.pair.Mie, "parser": _parse_mie},
+    }
+
+    nlist = hoomd.md.nlist.Cell(exclusions=["bond", "1-3"], buffer=nlist_buffer)
+
+    nbonded_forces = list()
+    for group in groups:
+        nbonded_forces.append(
+            atype_group_map[group]["parser"](
+                container=atype_group_map[group]["container"](nlist=nlist),
+                atypes=groups[group],
+                combining_rule=top.combining_rule,
+            )
+        )
+
+    return nbonded_forces
+
+
+def _parse_bond_forces(top, potential_types, potential_refs, base_units):
+    """Parse bond forces."""
+
+    def _parse_harmonic(container, btypes):
+        for btype in btypes:
+            # TODO: Unit conversion
+            container.params["-".join(btype.member_types)] = {
+                "k": btype.parameters["k"],
+                "r0": btype.parameters["r_eq"],
+            }
+        return container
+
+    unique_btypes = top.bond_types(filter_by=PotentialFilters.UNIQUE_NAME_CLASS)
+    groups = dict()
+    for btype in unique_btypes:
+        group = potential_types[btype]
+        if group not in groups:
+            groups[group] = [btype]
+        else:
+            groups[group].append(btype)
+
+    for group in groups:
+        expected_units_dim = potential_refs[group][
+            "expected_parameters_dimensions"
+        ]
+        groups[group] = _convert_params_units(
+            groups[group], expected_units_dim, base_units
+        )
+
+    btype_group_map = {
+        "HarmonicBondPotential": {
+            "container": hoomd.md.bond.Harmonic,
+            "parser": _parse_harmonic,
+        },
+    }
+    bond_forces = list()
+    for group in groups:
+        bond_forces.append(
+            btype_group_map[group]["parser"](
+                container=btype_group_map[group]["container"](),
+                btypes=groups[group],
+            )
+        )
+
+    return bond_forces
+
+
+def _parse_angle_forces(top, potential_types, potential_refs, base_units):
+    """Parse angle forces."""
+
+    def _parse_harmonic(container, agtypes):
+        for agtype in agtypes:
+            # TODO: Unit conversion
+            container.params["-".join(agtype.member_types)] = {
+                "k": agtype.parameters["k"],
+                "t0": agtype.parameters["theta_eq"],
+            }
+        return container
+
+    unique_agtypes = top.angle_types(
+        filter_by=PotentialFilters.UNIQUE_NAME_CLASS
+    )
+    groups = dict()
+    for agtype in unique_agtypes:
+        group = potential_types[agtype]
+        if group not in groups:
+            groups[group] = [agtype]
+        else:
+            groups[group].append(agtype)
+
+    base_units["angle"] = u.degree
```

Comment:
Does this mean the equilibrium angles will be in degrees? Hoomd uses radians for angles and dihedrals.

Reply:
Looks like we are getting the angles in degrees.

Here is what I get for a methane molecule using oplsaa

```
opls_140-opls_138-opls_140
_HOOMDDict{'t0': 107.80000000000646, 'k': 0.08411827276958087}
```

Hoomd is agnostic to units for distance and energy, but angles should definitely be in radians.

Reply:
Good call! I will fix it in the next commit.